src/generators/go/GoGenerator.ts

```diff
@@ -27,9 +29,17 @@ export class GoGenerator extends AbstractGenerator<GoOptions> {
 
   render(model: CommonModel, inputModel: CommonInputModel): Promise<RenderOutput> {
     const kind = TypeHelpers.extractKind(model);
-    if (kind === ModelKind.OBJECT) {
+    switch (kind) {
+    case ModelKind.OBJECT: {
       return this.renderStruct(model, inputModel);
     }
+    case ModelKind.ENUM: {
+      const typeName = model.$id && FormatHelpers.toPascalCase(model.$id, { transform: pascalCaseTransformMerge });
+      const result = `// ${typeName} represents an enum\ntype ${typeName} interface{}`;
+      return Promise.resolve(RenderOutput.toRenderOutput({ result, dependencies: [] }));
+    }
```

Comment:
It should be written as separate function. As I see you wanna make then the "true" enum in golang, not using the `interface{}`, yes?

Reply:
Also for this 

```ts
const result = `// ${typeName} represents an enum\ntype ${typeName} interface{}`;
```

Please use the `renderComments` and `renderBlock` functions.

Reply:
No I don't wanna make the enums to look right. This is not the final version of an enum as enums in go does not exist. There will need more work to make that happen.

For simplicity, I did not create a EnumRenderer since it is not rendering real enums.

Reply:
I know that there isn't real enum but you can create the "fake" enums like for string:

```go
type LeaveType string

const(
    AnnualLeave LeaveType = "AnnualLeave"
    Sick = "Sick"
    BankHoliday = "BankHoliday"
    Other = "Other"
)
```

I think that we go in this way.

Reply:
The point is that this PR is not meant for that. Its purpose is to enable the generator, and for that, I had to print some type on enums when generating the structs.

Reply:
There will be more issues to come for enabling all features in the Go Generator.

Reply:
OK, but really, please move the logic to separate function, not Renderer, and use the `renderComments` and `renderBlock` functions.

Reply:
I ended up creating the `EnumRenderer` and some tests around.